pkg/cloudprovider/provider/cherryservers/provider.go

```diff
@@ -0,0 +1,358 @@
+package cherryservers
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/cherryservers/cherrygo"
+
+	cloudprovidererrors "github.com/kubermatic/machine-controller/pkg/cloudprovider/errors"
+	"github.com/kubermatic/machine-controller/pkg/cloudprovider/instance"
+	cloudprovidertypes "github.com/kubermatic/machine-controller/pkg/cloudprovider/types"
+	"github.com/kubermatic/machine-controller/pkg/providerconfig"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/common"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+type provider struct {
+	configVarResolver *providerconfig.ConfigVarResolver
+}
+
+func New(configVarResolver *providerconfig.ConfigVarResolver) cloudprovidertypes.Provider {
+	return &provider{configVarResolver: configVarResolver}
+}
+
+type RawConfig struct {
+	TeamID    providerconfig.ConfigVarString `json:"teamID"`
+	ProjectID providerconfig.ConfigVarString `json:"projectID"`
+	Token     providerconfig.ConfigVarString `json:"token,omitempty"`
+	Plan      providerconfig.ConfigVarString `json:"plan"`
+	Location  providerconfig.ConfigVarString `json:"location"`
+	Labels    map[string]string              `json:"labels,omitempty"`
```

Comment:
If CherryServers doesn't support labels/tags, how are we handling labels defined here?

Reply:
CherryServers now supports Labels